apps/src/lib/client/tx.rs

```diff
@@ -1249,10 +1150,75 @@ pub async fn broadcast_tx(
 /// 3. The decrypted payload of the tx has been included on the blockchain.
 ///
 /// In the case of errors in any of those stages, an error message is returned
+#[cfg(not(feature = "ABCI"))]
 pub async fn submit_tx(
     address: TendermintAddress,
     to_broadcast: TxBroadcastData,
 ) -> Result<TxResponse, Error> {
+    // the data for finding the relevant events
+    let (_, wrapper_hash, decrypted_hash) = match &to_broadcast {
+        TxBroadcastData::Wrapper {
+            tx,
+            wrapper_hash,
+            decrypted_hash,
+        } => (tx, wrapper_hash, decrypted_hash),
+        _ => panic!("Cannot broadcast a dry-run transaction"),
```

Comment:
I can make the first change since that is indeed more explicit.

The pattern in the client code is generally to exit with a panic  when args are malformed. We return errors only upward if they stem at the program boundaries, i.e. socket connections etc.

Reply:
That makes sense, I think it'd be good to have less functions panic and instead propagate errors but it'd be a bigger thing to change